webapp/components/abstract-test-file-results-table.html

```diff
@@ -8,19 +8,73 @@
   <script>
     // eslint-disable-next-line no-unused-vars
     const AbstractTestFileResultsTable = superClass => class extends superClass {
-      static get is() {
-        return 'abstract-test-file-results-table';
-      }
-
       static get properties() {
         return {
           subtestNames: {
             type: Array,
             value: [],
           },
+          isTestHarness: {
+            type: Boolean,
+            value: false,
+          },
         };
       }
 
+      static get observers() {
+        return ['fetchTestFile(path, testRuns)'];
+      }
+
+      async fetchTestFile(path, testRuns) {
+        if (!path || !testRuns) {
+          return;
+        }
+
+        const resultPerTestRun = await Promise.all(
+          testRuns.map(tr => this.loadResultFile(tr)));
+
+        let isTestHarness = false;
+        const subtestNames = ['STATUS'];
+        resultPerTestRun.forEach((resultData, i) => {
+          if (!resultData) {
+            testRuns[i].subtests = {};
+            testRuns[i].subtests['STATUS'] = { status: '(results not found)' };
+            return;
+          }
+          testRuns[i].subtests = {};
+          testRuns[i].subtests['STATUS'] = { status: resultData.status };
+          if (['OK', 'ERROR'].includes(resultData.status)) {
+            isTestHarness = true;
+          }
+
+          for (let subtestResult of resultData.subtests) {
+            testRuns[i].subtests[subtestResult.name] = subtestResult;
+            if (!subtestNames.includes(subtestResult.name)) {
+              subtestNames.push(subtestResult.name);
+            }
+          }
+        });
+
+        this.isTestHarness = isTestHarness;
+        this.subtestNames = subtestNames;
+      }
+
+      async loadResultFile(testRun) {
+        const url = this.resultsURL(testRun, this.path);
+        const response = await window.fetch(url);
+        if (!response.ok) {
+          return null;
+        }
+        return response.json();
+      }
+
+      resultsURL(testRun, path) {
+        path = this.encodeTestPath(path);
+        // This is relying on the assumption that result files end with '-summary.json.gz'.
```

Comment:
Aside: I don't think anyone will be too sad if @mdittmer makes these lines go *poof* :)